### PR TITLE
fix to check that an end time is not later than the start time

### PR DESCRIPTION
**Pull request by kunmi02**
This fix is to check the the end date or the end time is not later than the end date or the end time while creating schedules, activity and talks

### CODE REVIEW

File: traveller/modules/schedule/view.py [hunk 1]
```diff
@@ -21,6 +26,12 @@
 globals()[mhelp.blueprint_str] = mhelp.blueprint
 module_blueprint = globals()[mhelp.blueprint_str]
 
+
+def _date_difference(start_date, end_date):
+    if end_date < start_date:
+        return True
```
Comment (ThiefMaster) [inline traveller/modules/schedule/view.py:32]: That function makes the code less readable. Just put the check in the places where you call it atm...

File: traveller/modules/schedule/view.py [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# import datetime
```
Comment (ThiefMaster) [inline traveller/modules/schedule/view.py:1]: committing commented-out code is usually not a good idea

**Comment by kunmi02** [comment]
@ThiefMaster Kindly review again

**Comment by Abdur-rahmaanJ** [comment]
@kunmi02 ill do it by tonight

File: traveller/modules/schedule/view.py [hunk 3]
```diff
@@ -50,7 +64,9 @@ def add_activity(year, day_id, act_type):
     if act_type == 'normal_activity':
         day = Day.query.get(day_id)
         form = NormalActivityForm()
-        form.validate()
```
Comment (jugmac00) [inline traveller/modules/schedule/view.py:53]: I do not think you can just delete `form.validiate` as there is more data than just the start and the end time.

**Review by jugmac00 — CHANGES_REQUESTED** [review]

File: traveller/modules/schedule/view.py [hunk 4]
```diff
@@ -21,6 +24,12 @@
 globals()[mhelp.blueprint_str] = mhelp.blueprint
 module_blueprint = globals()[mhelp.blueprint_str]
 
+
+def _date_difference(start_date, end_date):
+    if end_date < start_date:
+        return True
```
Comment (jugmac00) [inline traveller/modules/schedule/view.py:30]: Please remove unused code.

File: traveller/modules/schedule/view.py [hunk 5]
```diff
@@ -37,6 +46,9 @@ def add_day(year):
 
     form = DayForm()
     form.validate()
+    if date.today() < form.date.data:
+        flash(notify_danger("new schedule date should be today or later"))
+        return mhelp.redirect_url('y.schedule', year=year)
```
Comment (jugmac00) [inline traveller/modules/schedule/view.py:51]: This needs to be the other way round.

`if date.today() > form.date.data: `

File: traveller/modules/schedule/view.py [hunk 6]
```diff
@@ -51,6 +63,9 @@ def add_activity(year, day_id, act_type):
         day = Day.query.get(day_id)
         form = NormalActivityForm()
         form.validate()
+        if form.start_time.data < form.end_time.data:
+            flash(notify_danger("End time should be greater than start date"))
+            return mhelp.redirect_url('y.schedule', year=year)
```
Comment (jugmac00) [inline traveller/modules/schedule/view.py:68]: It is the other way round.

File: traveller/modules/schedule/view.py [hunk 7]
```diff
@@ -61,6 +76,9 @@ def add_activity(year, day_id, act_type):
         form = TalkActivityForm()
 
         form.validate()
+        if form.start_time.data < form.end_time.data:
+            flash(notify_danger("End time should be greater than start date"))
+            return mhelp.redirect_url('y.schedule', year=year)
```
Comment (jugmac00) [inline traveller/modules/schedule/view.py:81]: It is the other way round.

File: traveller/modules/schedule/view.py [hunk 8]
```diff
@@ -77,18 +95,25 @@ def edit_activity(year, act_id, act_type):
     if act_type == 'normal_activity':
         form = NormalActivityForm()
         form.validate()
+        if form.start_time.data < form.end_time.data:
+            flash(notify_danger("End time should be greater than start date"))
+            return mhelp.redirect_url('y.schedule', year=year)
```
Comment (jugmac00) [inline traveller/modules/schedule/view.py:100]: It is the other way round.

File: traveller/modules/schedule/view.py [hunk 9]
```diff
@@ -77,18 +95,25 @@ def edit_activity(year, act_id, act_type):
...
         activity = Activity.query.get(act_id)
         form.populate_obj(activity)
         activity.update()
     elif act_type == 'talk':
         day = Day.query.get(day_id)
         form = TalkActivityForm()
         form.validate()
+        if form.start_time.data < form.end_time.data:
+            flash(notify_danger("End date should be greater than start date"))
+            return mhelp.redirect_url('y.schedule', year=year)
```
Comment (jugmac00) [inline traveller/modules/schedule/view.py:110]: It is the other way round.

**Review by jugmac00 — CHANGES_REQUESTED** [review]
There are a couple of bugs in your PR.

If you do not write automatic tests, at least you should test your code manually, ie create an instance and try your changes!

Thank you!

**Comment by kunmi02** [comment]
@jugmac00 Thanks for the review and suggested changes. I have applied the changes and also tested them. Kindly review it.

File: traveller/modules/schedule/view.py (hunk 4 above)
Comment (jugmac00) [reply traveller/modules/schedule/view.py:30]: This unused function is still not deleted.

**Review by jugmac00 — CHANGES_REQUESTED** [review]
Please delete the unused function `_date_difference`, then we are good to go.

**Comment by kunmi02** [comment]
@jugmac00 The unused function has been removed.
Thanks for your time and patience.

Kindly review again, please

**Review by jugmac00 — APPROVED** [review]

**Pull request merged**